### PR TITLE
[Cleanup] Invoice Design Cards To Tabs

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes replacing the Invoice Design cards with tabs. Screenshot:

Screenshot 2024-03-06 at 13 48 00

Important note: I also found a bug with `sync_invoice_quote_columns` and rendering the `quote_product_columns` tab/card, so I resolved it as well. Previously, if you changed this prop, the card stayed non-rendered until you saved changes.

Let me know your thoughts.

### CODE REVIEW

**Comment by Civolilah** [comment]
@turbo124 As you recommended a new approach for organizing tabs on invoice design pages, I've successfully implemented it. Here is the solution. Screenshot:

Screenshot 2024-03-08 at 17 52 31

Let me know your thoughts.

File: src/pages/settings/invoice-design/pages/general-settings/components/CompanyAddress.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { useTranslation } from 'react-i18next';
 import { SortableVariableList } from './SortableVariableList';
 import { useCustomField } from '$app/components/CustomField';
 
-export function CompanyAddress() {
+export default function CompanyAddress() {
```
Comment (beganovich) [inline src/pages/settings/invoice-design/pages/general-settings/components/CompanyAddress.tsx:16]: I see we started default exporting the functions. Why?

**Review by beganovich — COMMENTED** [review]
Looks good, just need a tiny clarification.

File: src/pages/settings/invoice-design/pages/general-settings/components/CompanyAddress.tsx (hunk 1 above)
Comment (Civolilah) [reply src/pages/settings/invoice-design/pages/general-settings/components/CompanyAddress.tsx:16]: @beganovich The reason for doing this is actually lazy loading. I found that we had already implemented lazy loading on the main `InvoiceDesign` component and `CustomDesigns` components. However, they were only linked components. Once I made `ClientDetails` and the rest of them linked tabs, I decided to follow this pattern and implement lazy loading on those as well. I hope that you will agree with following the already implemented pattern.

**Comment by Civolilah** [comment]
@turbo124 As we discussed on Slack, I've found a very edge case where the Save button wasn't showing at all. I resolved it, and I hope that everything looks great on your end as well.

**Comment by Civolilah** [comment]
@turbo124 As we discussed on Slack, I've found a very edge case where the Save button wasn't showing at all. I resolved it, and I hope that everything looks great on your end as well.

**Comment by beganovich** [comment]
@turbo124 can you confirm updated code passes qa, thanks!

**Comment by turbo124** [comment]
@beganovich looks good!

**Review by beganovich — APPROVED** [review]

**Pull request merged**